### ISSUE TITLE
error on node v22.1.0

### ISSUE THREAD

**Issue by xlc** [polkadot-js/api#5880]
```
node:internal/event_target:1095
  process.nextTick(() => { throw err; });
                           ^

TypeError: Cannot read properties of null (reading 'length')
    at WebSocket.__internal__onSocketMessage (file:///Users/xiliangchen/projects/acala/chopsticks/node_modules/@polkadot/rpc-provider/ws/index.js:392:40)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:821:20)
    at WebSocket.dispatchEvent (node:internal/event_target:756:26)
    at fireEvent (node:internal/deps/undici/undici:10966:14)
    at websocketMessageReceived (node:internal/deps/undici/undici:10988:7)
    at ByteParser.run (node:internal/deps/undici/undici:11439:17)
    at ByteParser._write (node:internal/deps/undici/undici:11315:14)
    at writeOrBuffer (node:internal/streams/writable:564:12)
    at _write (node:internal/streams/writable:493:10)
    at Writable.write (node:internal/streams/writable:502:10)

Node.js v22.1.0
```

failed on this line

https://github.com/polkadot-js/api/blob/baaa27ba9e5d81eefe08914973bc58a65fe648f6/packages/rpc-provider/src/ws/index.ts#L482

the message is
```
MessageEvent {
  type: 'message',
  defaultPrevented: false,
  cancelable: false,
  timeStamp: 1240.624208
}
```

and `message.data` is null for some reason

to reproduce, run chopsticks on nodejs v22.1.0 
Original issue https://github.com/AcalaNetwork/chopsticks/issues/753

**Comment by TarikGul** [comment]
Not surprised node v22 is giving some issues, but also happy to start working on giving it support.

**Comment by TarikGul** [comment]
This will most likely not be a fast fix, but I will do my best to get it resolved soon!

**Comment by TarikGul** [comment]
This issue has been fixed with Nodejs 22.3.0 - The only PR i could trace this too was: https://github.com/nodejs/node/pull/52370/files

I went through a lot of rabbit holes to see what I could do to improve this locally, to not much avail. I would just recommend using 22.3.0 or above -  as the lower versions had intrinsic issues with the Global `WebSocket`.

closing for now.
